test/Dialect/FIRRTL/round-trip.mlir

```diff
@@ -0,0 +1,19 @@
+// RUN: circt-opt %s | circt-opt | FileCheck %s
+
+firrtl.circuit "Foo" {
+  firrtl.option @Platform {
+    firrtl.option_case @FPGA
+    firrtl.option_case @ASIC
+  }
+
+  firrtl.module private @DefaultTarget(in %clock: !firrtl.clock) {}
+  firrtl.module private @FPGATarget(in %clock: !firrtl.clock) {}
+  firrtl.module private @ASICTarget(in %clock: !firrtl.clock) {}
+
+  // CHECK-LABEL: firrtl.module @Foo
+  firrtl.module @Foo(in %clock: !firrtl.clock) {
+    // CHECK: %inst_clock = firrtl.instance_choice inst interesting_name @DefaultTarget or @FPGATarget if @Platform:@FPGA or @ASICTarget if @Platform:@ASIC(in clock: !firrtl.clock)
+    %inst_clock = firrtl.instance_choice inst interesting_name @DefaultTarget or @FPGATarget if @Platform:@FPGA or @ASICTarget if @Platform:@ASIC(in clock: !firrtl.clock)
+    firrtl.strictconnect %inst_clock, %clock : !firrtl.clock
```

Comment:
The custom print/parse is the least important part. Nittily consider using something more like a dictionary as opposed to a sequence of `or`/`if`:

``` mlir
%inst_clock = firrtl.instance_choice inst @DefaultTarget alternatives {@Platform::@FPGA -> @FPGATarget, @ASICTarget -> @ASICTarget}
```